### PR TITLE
Update how SimpleSharedMacTestSuiteIT is executed

### PR DESCRIPTION
**Pull request by ctubbsii**
* Revert d21fc1bb4124df757e5d0691fd19281655d75761 for PR #5852, since #5851 can be fixed with this change instead, without adding a new test to the sunny profile; comprehensive table operations testing is outside the scope of the basic test case that the sunny profile should be limited to doing
* Add the `failIfNoTests = false` option to the `@Suite` annotation so that it does not fail if some combination of options fails to result in any matching test cases to run (minicluster is still started for the test, though no tests will run; that is addressed by the next change below)
* Explicitly run only the regular junit-jupiter engine, and not the junit-platform-suite engine, by default when running the sunny profile, so that it does not execute SimpleSharedMacTestSuiteIT's BeforeSuite and AfterSuite methods, which is a waste of time since that suite does not have any tests with the SunnyDay tag

Also:
* Move the class so it fits into the module/package naming conventions
* Add a CI check for module/package naming conventions for future work

More background:

  The maven-surefire-plugin/maven-failsafe-plugin's JUnit5 provider will use both the normal (non-suite) `junit-jupiter` engine and the `junit-platform-suite` engine. These engines behave differently when handling the surefire/failsafe option to filter tests based on groups. The regular engine will filter tests based on the configured groups by the `@Tag` annotation before executing the tests. The suite engine appears to identify the suites to run first, then filters the test cases contained in that suite by the `@Tag` annotation. While the regular engine's failure to find any tests can be handled by the Maven properties, `(surefire|failsafe).failIfNoSpecifiedTests`, that does not work for the suite engine. Instead, the suite engine requires `@Suite(failIfNoTests = false)` to be used. That will prevent the failure to find tests from failing the build, but it will still execute the suite's `@BeforeSuite` and `@AfterSuite` methods. An additional configuration is needed to explicitly control which engines are executed by surefire/failsafe in order to avoid executing "empty" suites' before/after steps entirely. In Accumulo, we do not have any suite tests that need to run during the `sunny` profile, so we can safely skip the suite engine's execution when activating that profile. We do not want to exclude the engine, though, because that limits the user's ability to override settings on the command-line with an exclude option. So instead, we limit the set of included engines to only the regular one, so users can still override the include/exclude properties to have more granular control, if they wish.

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/suites/SimpleSharedMacTestSuiteIT.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
  * IMPORTANT NOTE: Only the concrete classes should be marked with "IT_SimpleSuite". For example,
  * marking an abstract class will not add its implementations to the suite.
  */
-@Suite
+@Suite(failIfNoTests = false)
```
Comment (dlmarion) [inline test/src/main/java/org/apache/accumulo/test/suites/SimpleSharedMacTestSuiteIT.java:46]: I'm not sure we should make this change. We are excluding the junit-platform-suite engine during the sunny profile, so this should not run. With this change the test will succeed if no tests are found during the non-sunny test runs.

File: src/build/ci/check-module-package-conventions.sh [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+# under the License.
+#
+
+# The purpose of this ci script is to ensure that a pull request
+# doesn't unintentionally add a jar resource that would cause
+# problems with jar sealing by breaking our package naming
+# conventions, which are to use package names based on the
+# module, so they are unique.
+NUM_EXPECTED=0
+ALLOWED=(
```
Comment (dlmarion) [inline src/build/ci/check-module-package-conventions.sh:27]: why not just allow everything in src/main/resources and src/test/resources? Do we need to enumerate here?

**Comment by dlmarion** [comment]
> The suite engine appears to identify the suites to run first, then filters the test cases contained in that suite by the `@Tag` annotation.

This makes sense now why it did not work when I tried to exclude the suite class itself using a tag in the sunny profile.

File: pom.xml [hunk 3]
```diff
@@ -1529,6 +1529,13 @@
       <id>sunny</id>
       <properties>
         <failsafe.groups>SunnyDay</failsafe.groups>
+        <!--
+          exclude junit-platform-suite engine from this list of engines to run,
+          so it skips the SimpleSuite tests, which contain no sunny tests, to
+          prevent the suite engine from running the BeforeSuite and AfterSuite
+          minicluster setup and teardown unnecessarily for zero tests
+        -->
+        <failsafe.includeJUnit5Engines>junit-jupiter</failsafe.includeJUnit5Engines>
```
Comment (kevinrr888) [inline pom.xml:1538]: Won't this be a problem if a SimpleSuite test is added to sunny in the future? Maybe we are saying a SimpleSuite test can not be a sunny test (which is fine as including a test as SimpleSuite is more of thing that's nice to include when we can). If this is the case, this should be added to the description for the `SimpleSharedMacTestSuiteIT` class

File: test/src/main/java/org/apache/accumulo/test/suites/SimpleSharedMacTestSuiteIT.java (hunk 1 above)
Comment (ctubbsii) [reply test/src/main/java/org/apache/accumulo/test/suites/SimpleSharedMacTestSuiteIT.java:46]: While we are excluding it for our sunny profile, this error can occur with any combination of command-line options that the user might set that results in filtering out the tests down to an empty set. Since there's no way that I can see to pass user config options to this suite to disable the default failure behavior, we have to leave this here, so we don't restrict what kinds of testing command-line options the user can set.

The output will show a success, but it will also show that 0 tests were run. If we remove this, then users won't even have the option of executing certain combinations of tests, because it will always fail. Leaving it in was my compromise. Perhaps we can request a feature upstream in junit/surefire to be able to control the behavior of the suite engine with a command-line option?

File: src/build/ci/check-module-package-conventions.sh (hunk 2 above)
Comment (ctubbsii) [reply src/build/ci/check-module-package-conventions.sh:27]: We need to enumerate because of the way I wrote the script's grep behavior... copied from other similar CI scripts. Allowing wildcard or pattern parsing gets very tricky.

Most of these should get cleaned up with #5867 , so I think it's better to have them enumerated for now.

File: pom.xml (hunk 3 above)
Comment (ctubbsii) [reply pom.xml:1538]: Yes it would be a problem if we did that. But, if we did that, we can just remove this line. This exists only to avoid starting and stopping a minicluster for zero tests. If we add a sunny test to the SimpleSuite, then we wouldn't have zero tests, and then it would be okay to have it started. That's why I put this comment here.

Comment (kevinrr888) [reply pom.xml:1538]: My concern would be that it may be hard for a dev to tell that this line needs to be removed if a sunny SimpleSuite is added. Would have to notice that the SimpleSuite test is not run when running sunny, which may go unnoticed.

Maybe just updating the `SimpleSharedMacTestSuiteIT` class comment to point to this maven config would be sufficient.

Comment (ctubbsii) [reply pom.xml:1538]: If somebody is modifying the sunny profile to add remove tests or alter its default behavior in some way, then this is the right place to look, so I think this comment is sufficient. The behavior of the sunny profile is not specific to `SimpleSharedMacTestSuiteIT`, so putting it there is not the right place for any further comments. I'm also disinclined to add more details that are just as likely to confuse and mislead as they are to help, because so much of the behavior depends on precisely which command-line options are used by the user. To explain all the possibilities of what a developer might need to change under various different options and desired behaviors, could fill a book.
